### PR TITLE
fix self-edge bug in GCN and GAT.

### PR DESCRIPTION
**Pull request by sufeidechabei**
## Description
I find that  some nodes in citation dataset have self-edge orginally, if we use `g.add_edges(g.nodes(), g.nodes())` directly, some nodes will have more than one self-edge.  This PR fixes that.

### CODE REVIEW

**Comment by mufeili** [comment]
I just took a look and it seems that `data.graph` does not have self edges. Have you verified the issue yourself?

File: examples/mxnet/gat/gat_batch.py [hunk 1]
```diff
@@ -178,9 +178,12 @@ def main(args):
         test_mask = test_mask.as_in_context(ctx)
         val_mask = val_mask.as_in_context(ctx)
     # create graph
-    g = DGLGraph(data.graph)
+    g = data.graph
     # add self-loop
+    g.remove_edges_from(g.selfloop_edges())
+    g = DGLGraph(g)
     g.add_edges(g.nodes(), g.nodes())
+
```
Comment (mufeili) [inline examples/mxnet/gat/gat_batch.py]: One empty line only to be consistent with the rest of the file.

File: examples/pytorch/gcn/gcn_mp.py [hunk 2]
```diff
@@ -147,10 +147,13 @@ def main(args):
...
     # graph preprocess and calculate normalization factor
-    g = DGLGraph(data.graph)
-    n_edges = g.number_of_edges()
+    g = data.graph
+    g.remove_edges_from(g.selfloop_edges())
+    g = DGLGraph(g)
     # add self loop
     g.add_edges(g.nodes(), g.nodes())
+    n_edges = g.number_of_edges()
+    # create model
```
Comment (mufeili) [inline examples/pytorch/gcn/gcn_mp.py]: Remove L156, which seems to be wrong.

**Comment by sufeidechabei** [comment]
> Looks good overall. Some minor comments.

Done!

**Comment by yzh119** [comment]
Why more than one self-edge matters? AFAIN self edge does not affect Laplacian matrix, does DGL have problem handling multi-edge?

**Comment by mufeili** [comment]
@yzh119 , see below for your reference.

```
import torch
from dgl import DGLGraph

g = DGLGraph()
g.add_nodes(1)
g.add_edges([0, 0], [0, 0])
g.ndata['h'] = torch.tensor([[1.]])
g.update_all(fn.copy_src('h', 'm'), fn.sum('m', 'h'))
```

**Comment by shengwenLeong** [comment]
In the tutorial of GCN in the document, the code didn't contain adding the self-edge operation. so should we fix it?

**Comment by mufeili** [comment]
@shengwenLeong Nice catch! Thank you for the notice.

@HQ01 Could you please help fixing it?

**Comment by HQ01** [comment]
@shengwenLeong tutorial has been fixed

**Pull request merged**